# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/libserial/hotfix-opt-doxygen.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- libserial-1.0.0/CMakeLists.txt.vanilla	2025-06-27 16:53:35.647753412 +0200
+++ libserial-1.0.0/CMakeLists.txt	2025-06-27 16:55:40.684981490 +0200
@@ -42,7 +42,7 @@
 # be set before call to FIND_PACKAGE(Threads).
 #
 SET(THREADS_HAVE_PTHREAD_ARG 1)
-FIND_PACKAGE(Doxygen REQUIRED)
+FIND_PACKAGE(Doxygen)
 FIND_PACKAGE(Boost COMPONENTS unit_test_framework REQUIRED)
 FIND_PACKAGE(PythonLibs REQUIRED)
 #FIND_PACKAGE(SIP REQUIRED)
@@ -157,8 +157,10 @@
   ${CMAKE_CURRENT_BINARY_DIR}/doxygen.conf.in @ONLY
   )
 
+if(DOXYGEN_FOUND)
 ADD_CUSTOM_TARGET(docs ALL
   ${DOXYGEN_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/doxygen.conf.in
   WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
   COMMENT "Generating API documentation with Doxygen" VERBATIM
 )
+endif()
